### PR TITLE
[Merged by Bors] - feat: a reflexive vector space (in the purely algebraic sense) is finite dimensional

### PR DESCRIPTION
**Pull request by ocfnash**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by eric-wieser** [comment]
!bench

File: Mathlib/LinearAlgebra/Dual.lean [hunk 1]
```diff
@@ -650,6 +651,24 @@ instance _root_.MulOpposite.instModuleIsReflexive : IsReflexive R (MulOpposite M
 instance _root_.ULift.instModuleIsReflexive.{w} : IsReflexive R (ULift.{w} M) :=
   equiv ULift.moduleEquiv.symm
 
+instance instFiniteDimensionalOfIsReflexive
+    (K : Type uK) (V : Type uV) [Field K] [AddCommGroup V] [Module K V] [IsReflexive K V] :
+    FiniteDimensional K V := by
+  rw [FiniteDimensional, ← rank_lt_aleph0_iff]
+  by_contra contra
```
Comment (Ruben-VandeVelde) [inline Mathlib/LinearAlgebra/Dual.lean:658]:
```suggestion
  by_contra! contra
```

Comment (ocfnash) [reply Mathlib/LinearAlgebra/Dual.lean]: Thanks!

**Pull request closed without merging**